test/SpokePoolClient.deposits.ts

```diff
@@ -1,83 +1,105 @@
 import { SpokePoolClient } from "../src/clients";
+import { Deposit } from "../src/interfaces";
 import {
   Contract,
   SignerWithAddress,
+  buildFill,
   createSpyLogger,
   deploySpokePoolWithToken,
   destinationChainId,
-  enableRoutes,
   ethers,
   expect,
   originChainId,
   setupTokensForWallet,
-  simpleDeposit,
+  toBNWei,
 } from "./utils";
 
 let spokePool: Contract, erc20: Contract, destErc20: Contract, weth: Contract;
-let depositor1: SignerWithAddress, depositor2: SignerWithAddress;
+let depositor: SignerWithAddress, relayer1: SignerWithAddress, relayer2: SignerWithAddress;
 let deploymentBlock: number;
-const destinationChainId2 = destinationChainId + 1;
+
+const originChainId2 = originChainId + 1;
 
 let spokePoolClient: SpokePoolClient;
 
-describe("SpokePoolClient: Deposits", async function () {
+describe("SpokePoolClient: Fills", async function () {
   beforeEach(async function () {
-    [, depositor1, depositor2] = await ethers.getSigners();
-    ({ spokePool, erc20, destErc20, weth, deploymentBlock } = await deploySpokePoolWithToken(originChainId));
-    await enableRoutes(spokePool, [{ originToken: erc20.address, destinationChainId: destinationChainId2 }]);
-    spokePoolClient = new SpokePoolClient(createSpyLogger().spyLogger, spokePool, null, originChainId, deploymentBlock);
+    [, depositor, relayer1, relayer2] = await ethers.getSigners();
+    ({ spokePool, erc20, destErc20, weth, deploymentBlock } = await deploySpokePoolWithToken(
+      originChainId,
+      destinationChainId
+    ));
+    await spokePool.setChainId(destinationChainId); // The spoke pool for a fill should be at the destinationChainId.
+
+    spokePoolClient = new SpokePoolClient(
+      createSpyLogger().spyLogger,
+      spokePool,
+      null,
+      destinationChainId,
+      deploymentBlock
+    );
```

Comment:
I'm curious about this setup - why is the SpokePool deployed with chainId = `originChainId`, and then later switched to `destinationChainId` ? The `deploySpokePoolWithToken()` helper does some route-based setup, so adjusting the chainIds might have side-effects.

Reply:
This was refactored directly from the relayer. Should we follow up in a separate PR?